crates/ruff_python_semantic/src/red_knot/symbol_table/symbol.rs

```diff
@@ -0,0 +1,66 @@
+use crate::name::Name;
+use crate::red_knot::symbol_table::definition::Definition;
+use crate::red_knot::symbol_table::ScopeId;
+use bitflags::bitflags;
+use ruff_db::vfs::VfsFile;
+use ruff_index::newtype_index;
+
+#[derive(Eq, PartialEq, Debug)]
+pub struct Symbol {
+    pub(super) name: Name,
+    pub(super) flags: SymbolFlags,
+    pub(super) scope: ScopeId,
+    pub(super) definitions: smallvec::SmallVec<[Definition; 1]>,
```

Comment:
@carljm Is there a particular reason why the existing semantic model stores the definitions in a `FxHashMap<SymbolId, Vec<Definition>>` rather than on the symbol itself?

Reply:
I did it that way in order to keep the symbol table potentially separable from definitions and thus from the AST.